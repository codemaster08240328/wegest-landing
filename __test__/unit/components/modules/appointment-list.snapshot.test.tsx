import renderer from 'react-test-renderer'
import AppointmentList from '@/components/modules/appointment-list'

describe('<AppointmentList> snapshot', () => {
  const appointments = [
    {
      id: 'iron-man',
      start: '08:25',
      end: '08:50',
      customer: 'Tony Stark',
      service: 'Hair Cut',
      color: 'darkgreen',
      onClick: () => null,
    },
    {
      id: 'black-widow',
      start: '08:55',
      end: '09:20',
      customer: 'Natasha Romanoff',
      service: 'Hair Treatment',
      color: 'darkgray',
      onClick: () => null,
    },
    {
      id: 'hulk',
      start: '09:20',
      end: '09:45',
      customer: 'Bruce Banner',
      service: 'Beard Trim',
      color: 'slategray',
      onClick: () => null,
    },
  ]

  it('renders properly with all required props', () => {
    expect(
      renderer.create(<AppointmentList appointments={appointments} />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders with a class', () => {
    expect(
      renderer
        .create(
          <AppointmentList
            className='class-test'
            appointments={appointments}
          />,
        )
        .toJSON(),
    ).toMatchSnapshot()
  })
})
