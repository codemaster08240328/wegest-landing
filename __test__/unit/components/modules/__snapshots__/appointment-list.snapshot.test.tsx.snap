// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentList> snapshot renders properly with all required props 1`] = `
<div
  className="makeStyles-root-1 "
  data-testid="appointments"
>
  <div
    className="makeStyles-root-3 "
  >
    <div
      className="makeStyles-timeSlot-4"
      data-testid="timeSlot"
      style={
        Object {
          "background": "darkgreen",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        08:25
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        08:50
      </span>
    </div>
    <div
      className="makeStyles-appointment-5"
      data-testid="appointment"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Tony Stark
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        Hair Cut
      </span>
    </div>
    <div
      className="makeStyles-action-6"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root Component-root-7 Component-root-8"
        data-testid="el-icon-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root material-icons-outlined"
            data-testid="el-icon"
          >
            info
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-3 "
  >
    <div
      className="makeStyles-timeSlot-4"
      data-testid="timeSlot"
      style={
        Object {
          "background": "darkgray",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        08:55
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        09:20
      </span>
    </div>
    <div
      className="makeStyles-appointment-5"
      data-testid="appointment"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Natasha Romanoff
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        Hair Treatment
      </span>
    </div>
    <div
      className="makeStyles-action-6"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root Component-root-7 Component-root-9"
        data-testid="el-icon-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root material-icons-outlined"
            data-testid="el-icon"
          >
            info
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-3 "
  >
    <div
      className="makeStyles-timeSlot-4"
      data-testid="timeSlot"
      style={
        Object {
          "background": "slategray",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        09:20
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        09:45
      </span>
    </div>
    <div
      className="makeStyles-appointment-5"
      data-testid="appointment"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Bruce Banner
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        Beard Trim
      </span>
    </div>
    <div
      className="makeStyles-action-6"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root Component-root-7 Component-root-10"
        data-testid="el-icon-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root material-icons-outlined"
            data-testid="el-icon"
          >
            info
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<AppointmentList> snapshot renders with a class 1`] = `
<div
  className="makeStyles-root-1 class-test"
  data-testid="appointments"
>
  <div
    className="makeStyles-root-3 "
  >
    <div
      className="makeStyles-timeSlot-4"
      data-testid="timeSlot"
      style={
        Object {
          "background": "darkgreen",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        08:25
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        08:50
      </span>
    </div>
    <div
      className="makeStyles-appointment-5"
      data-testid="appointment"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Tony Stark
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        Hair Cut
      </span>
    </div>
    <div
      className="makeStyles-action-6"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root Component-root-7 Component-root-11"
        data-testid="el-icon-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root material-icons-outlined"
            data-testid="el-icon"
          >
            info
          </span>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-3 "
  >
    <div
      className="makeStyles-timeSlot-4"
      data-testid="timeSlot"
      style={
        Object {
          "background": "darkgray",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        08:55
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        09:20
      </span>
    </div>
    <div
      className="makeStyles-appointment-5"
      data-testid="appointment"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Natasha Romanoff
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        Hair Treatment
      </span>
    </div>
    <div
      className="makeStyles-action-6"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root Component-root-7 Component-root-12"
        data-testid="el-icon-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root material-icons-outlined"
            data-testid="el-icon"
          >
            info
          </span>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    className="makeStyles-root-3 "
  >
    <div
      className="makeStyles-timeSlot-4"
      data-testid="timeSlot"
      style={
        Object {
          "background": "slategray",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        09:20
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        09:45
      </span>
    </div>
    <div
      className="makeStyles-appointment-5"
      data-testid="appointment"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Bruce Banner
      </p>
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        Beard Trim
      </span>
    </div>
    <div
      className="makeStyles-action-6"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root Component-root-7 Component-root-13"
        data-testid="el-icon-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root material-icons-outlined"
            data-testid="el-icon"
          >
            info
          </span>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
